### PR TITLE
Refactored grammar AST export, renamed naming.ts

### PR DESCRIPTION
**Pull request by spoenemann**
This PR further cleans up our public API.

 * Instead of exporting all grammar AST elements directly to the package index, I put them into the `GrammarAST` namespace.
 * Removed a few files from the package index because they are not required by most downstream projects.
 * Renamed `naming.ts` to `name-provider.ts` for consistency.

### CODE REVIEW

File: packages/langium-cli/src/generate.ts [hunk 1]
```diff
@@ -6,8 +6,7 @@
 
 import fs from 'fs-extra';
 import {
-    AbstractRule, createLangiumGrammarServices, getDocument, Grammar, Interface, isAbstractRule, isGrammar, isInterface, isParserRule, isType,
-    LangiumDocument, linkContentToContainer, Type
+    createLangiumGrammarServices, getDocument, Grammar, GrammarAST, LangiumDocument, linkContentToContainer
 } from 'langium';
```
Comment (Lotes) [inline packages/langium-cli/src/generate.ts:10]: I like the prefixing idea 'GrammarAST.<...>'. But the naming is not very 1:1. I mean `LangiumAST` would be more direct, whereas `GrammarAST` does not tell me what grammar is meant, or?

Grammar -> LangiumGrammar
GrammarAST -> LangiumAST
LangiumDocument -> perfect

Comment (spoenemann) [reply packages/langium-cli/src/generate.ts:10]: Right, but then I'd name it `LangiumGrammarAST`. LangiumAST is still ambiguous.

Comment (pluralia) [reply packages/langium-cli/src/generate.ts:10]: Isn't `LangiumGrammarAST` too long?

**Review by montymxb — COMMENTED** [review]
When trying to build locally this causes the examples to no longer generate. It seems that the imports need to updated in the CLI as well. The specific error I'm seeing is `Cannot find module 'langium/src/grammar/type-system'` in **langium-cli/src/generator/ast-generator.ts**. Taking a look there it seems the import just needs to be updated to reflect these changes.

File: packages/langium-cli/src/generate.ts (hunk 1 above)
Comment (spoenemann) [reply packages/langium-cli/src/generate.ts:10]: @pluralia you are right, too. So then I'd leave the namespace as `GrammarAST` for now. We should rethink our exported API on a more general level. Ideas so far:

 - Create focused index files in every subfolder, narrow the main index (https://github.com/langium/langium/discussions/155)
 - Create a hierarchy of namespaces like the [vscode API](https://code.visualstudio.com/api/references/vscode-api)
 - Keep the current global index and ensure proper naming of exported things

**Review by montymxb — COMMENTED** [review]
Build issues corrected with cli ✅ . No issues here with merging, will defer to remaining reviewers.

**Review by Lotes — APPROVED** [review]

**Review by pluralia — APPROVED** [review]

File: packages/langium-cli/src/generate.ts (hunk 1 above)
Comment (pluralia) [reply packages/langium-cli/src/generate.ts:10]: Should we put these ideas in a separate issue?

Comment (spoenemann) [reply packages/langium-cli/src/generate.ts:10]: I don't know whether it's really an issue. I'd make a discussion and turn it into an issue if we find problems with the current approach.

**Pull request merged**